homeassistant/components/isy994/entity.py

```diff
@@ -87,57 +79,47 @@ def device_info(self) -> DeviceInfo | None:
             basename = node.name
 
-        if hasattr(node, "parent_node") and node.parent_node is not None:
-            # This is not the parent node, get the parent node.
-            node = node.parent_node
-            basename = node.name
+        if hasattr(node, "parent_node"):  # Verify this is a Node class
+            if node.parent_node is not None:
+                # This is not the parent node, get the parent node.
+                node = node.parent_node
+                basename = node.name
+        else:
+            # Default to the hub device if parent node is not a physical device
+            return DeviceInfo(identifiers={(DOMAIN, uuid)})
 
         device_info = DeviceInfo(
-            manufacturer="Unknown",
-            model="Unknown",
-            name=basename,
+            identifiers={(DOMAIN, f"{uuid}_{node.address}")},
+            manufacturer=node.protocol,
+            name=f"{basename} ({(str(node.address).rpartition(' ')[0] or node.address)})",
             via_device=(DOMAIN, uuid),
             configuration_url=url,
+            suggested_area=node.folder,
         )
 
-        if hasattr(node, "address"):
-            assert isinstance(node.address, str)
-            device_info[
-                ATTR_NAME
-            ] = f"{basename} ({(node.address.rpartition(' ')[0] or node.address)})"
-        if hasattr(node, "primary_node"):
-            device_info[ATTR_IDENTIFIERS] = {(DOMAIN, f"{uuid}_{node.address}")}
-        # ISYv5 Device Types
-        if hasattr(node, "node_def_id") and node.node_def_id is not None:
-            model: str = str(node.node_def_id)
-            # Numerical Device Type
-            if hasattr(node, "type") and node.type is not None:
-                model += f" {node.type}"
-            device_info[ATTR_MODEL] = model
-        if hasattr(node, "protocol"):
-            model = str(device_info[ATTR_MODEL])
-            manufacturer = str(node.protocol)
-            if node.protocol == PROTO_ZWAVE:
-                # Get extra information for Z-Wave Devices
-                manufacturer += f" MfrID:{node.zwave_props.mfr_id}"
-                model += (
-                    f" Type:{node.zwave_props.devtype_gen} "
-                    f"ProductTypeID:{node.zwave_props.prod_type_id} "
-                    f"ProductID:{node.zwave_props.product_id}"
-                )
-            device_info[ATTR_MANUFACTURER] = manufacturer
-            device_info[ATTR_MODEL] = model
-        if hasattr(node, "folder") and node.folder is not None:
-            device_info[ATTR_SUGGESTED_AREA] = node.folder
-        # Note: sw_version is not exposed by the ISY for the individual devices.
+        # ISYv5 Device Types can provide model and manufacturer
+        model: str = "Unknown"
+        if node.node_def_id is not None:
+            model = str(node.node_def_id)
+
+        # Numerical Device Type
+        if node.type is not None:
+            model += f" ({node.type})"
```

Comment:
This looks like the model could start with `Unknown`. Is that intended?

Reply:
Yes, that's intentional. If it's not a Z-Wave device or current generation ISY firmware, you might only have the Insteon type codes, so it would show up as `Unknown (16.4.2.0)` with `Insteon` as the manufacturer.

Reply:
Ok.